snitun/server/listener_sni.py

```diff
@@ -205,8 +205,13 @@ async def start(
             with suppress(MultiplexerTransportError):
                 await multiplexer.delete_channel(channel)
 
-        except (MultiplexerTransportError, OSError, RuntimeError, ConnectionResetError):
-            _LOGGER.debug("Transport closed by Proxy for %s", channel.id)
+        except (
+            MultiplexerTransportError,
+            OSError,
+            RuntimeError,
+            ConnectionResetError,
+        ) as exc:
+            _LOGGER.debug("Transport closed by Proxy for %s: %s", channel.id, exc)
```

Comment:
Log the exception that causes the connection to close instead of throwing it away here